lib/compilers/racket.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2022, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import {CompilationResult, ExecutionOptions} from '../../types/compilation/compilation.interfaces';
+import {ParseFilters} from '../../types/features/filters.interfaces';
+import {BaseCompiler} from '../base-compiler';
+import {logger} from '../logger';
+
+export class RacketCompiler extends BaseCompiler {
+    private raco: string;
+
+    static get key() {
+        return 'racket';
+    }
+
+    constructor(info, env) {
+        // Disable output filters, as they currently don't do anything
+        if (!info.disabledFilters) {
+            info.disabledFilters = ['labels', 'directives', 'commentOnly', 'trim'];
+        }
+        super(info, env);
+        this.raco = this.compilerProps(`compiler.${this.compiler.id}.raco`);
+    }
+
+    override optionsForFilter(filters: ParseFilters, outputFilename: any, userOptions?: any): string[] {
+        // We currently always compile to bytecode first and then decompile.
+        // Forcing `binary` on like this ensures `objdump` will be called for
+        // the decompilation phase.
+        filters.binary = true;
+
+        return [];
+    }
+
+    override supportsObjdump(): boolean {
+        return true;
+    }
+
+    protected override getSharedLibraryPathsAsArguments(libraries: any, libDownloadPath?: any): string[] {
```

Comment:
The base class has more specific types for this function, can you add those please? :)

Reply:
Hmm, for this one, I don't see the new types in `base-compiler`:

https://github.com/compiler-explorer/compiler-explorer/blob/f22a2a3a782a22e820ae57f3daa755c55347fbd0/lib/base-compiler.ts#L748

...but maybe you want me to copy from some of the newer compilers, like this one...?

https://github.com/compiler-explorer/compiler-explorer/blob/f22a2a3a782a22e820ae57f3daa755c55347fbd0/lib/compilers/toit.ts#L51

Reply:
That's super weird, I recall having added the types at some point. Nevermind then